eZ/Publish/API/Repository/TransactionService.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository;
+
+use Closure;
+
+interface TransactionService
+{
+    /**
+     * Begin transaction.
+     *
+     * Begins an transaction, make sure you'll call commit or rollback when done,
+     * otherwise work will be lost.
+     */
+    public function beginTransaction(): void;
+
+    /**
+     * Commit transaction.
+     *
+     * Commit transaction, or throw exceptions if no transactions has been started.
+     *
+     * @throws \RuntimeException If no transaction has been started
+     */
+    public function commit(): void;
+
+    /**
+     * Rollback transaction.
+     *
+     * Rollback transaction, or throw exceptions if no transactions has been started.
+     *
+     * @throws \RuntimeException If no transaction has been started
+     */
+    public function rollback(): void;
+
+    /**
+     * Executes a function in a transaction.
+     *
+     * If an exception occurs during execution of the function or transaction commit,
+     * the transaction is rolled back and the exception re-thrown.
```

Comment:
```suggestion
     * the transaction is rolled back and the exception is thrown again.
```

Reply:
IMO "re-throwing" an exception is more clear and conveys the fact that it is not wrapped with an additional exception or anything like that.